File: examples/floatingpoint/javascript/writer.js
#!/usr/bin/env node

'use strict';

var fs = require('fs'),
    yaml = require('js-yaml');

var BinParser = require('../../../javascript/index');

function main() {
  var parser = new BinParser.BinWriter(
    yaml.load(fs.readFileSync('../float.yml')),
    yaml.load(fs.readFileSync('../structure.yml')),
    yaml.load(fs.readFileSync('../types.yml')),
    {});

    process.stdout.write(parser.data);
}

// Wait for the stdout buffer to drain, see
// https://github.com/eslint/eslint/issues/317
process.on('exit', function() {
  process.reallyExit(main());
});

File: examples/floatingpoint/python/writer.py
#!/usr/bin/env python
import sys

import yaml

from bin_parser import BinWriter


parser = BinWriter(
    yaml.safe_load(open('../float.yml')),
    yaml.safe_load(open('../structure.yml')),
    yaml.safe_load(open('../types.yml')))
sys.stdout.write(parser.data)
